### DISCUSSION TITLE
Significance of tilde?

### DISCUSSION THREAD

**Discussion by kaby76** [pest-parser/pest#621] [General]
Pest uses tilde (`~`) between symbols and literals in grammars. So, instead of this which we would see in a typical EBNF-based grammar

    expression =  term (infix_operator term)*

we see this in Pest

    expression =  { term ~ (infix_operator ~ term)* }

Is there significance to the tilde operator in a generated parser? Most parser generators use whitespace to denote the end of a non-terminal name. Iso14977 allows spaces within non-terminal names but avoids tilde with two passes to parse the grammar in order to recognize non-terminals with gap.

**Comment by Abdillah** [comment]
I think the tilde symbol ease up the parsing, it acts as infix operator. It's converted into an `Expr::Seq(lhs, rhs)`. Well, it just passed through my mind when reading the code though, not so sure myself.

**Comment by gzz2000** [comment]
I also came across this question in my mind. I think we can eliminate the tilde. That way we would not introduce any ambiguity, and we can make writing grammar much more comfortable.
